src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,116 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Storage class allows the user to load/save
+ * all its tasks and their respective states in a txt file.
+ * The txt file will be created if it does not exist in the filePath
+ *
+ * @author Gerald Teo Jin Wei
+ * @version 0.1
+ * @since 2022-08-28
+ */
+public class Storage {
+
+    private String filePath;
+
+    /**
+     * Constructor for Storage class
+     * @param filePath Path to txt file that stores task list data
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * This method gets the existing txt file in the
+     * file path and creates the txt file if it
+     * does not exist
+     * @return File This is the txtfile that stores all the user's task information
+     */
+    public File getFile() {
+        File file = new File(filePath);
+        try {
+            if (file.createNewFile()) {
+                System.out.println("File is created!");
+            } else {
+                System.out.println("File already exists.");
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return file;
+    }
+
+    /**
+     * This method gets the existing text file and
+     * converts the text into a list of the user's tasks
+     * @return List This is the current list of the user's tasks
+     */
+    public List<Task> load() {
+        File file = this.getFile();
+        List<Task> taskList = new ArrayList<>();
+        try {
+            Scanner sc = new Scanner(file);
+            while (sc.hasNextLine()) {
+                String line = sc.nextLine();
+                String[] splitStr = line.split("\\|");

```

Comment:
The way this is implemented seems to inconvenience users wanting to use the `|` symbol in their task description.